botnim/collect_sources.py

```diff
@@ -32,22 +35,72 @@ def collect_sources_google_spreadsheet(context_name, source):
                         content += f'{header}:\n{row[header]}\n\n'
                     else:
                         content += f'{row[header]}\n\n'
+        
         if content:
-            file_streams.append((f'{context_name}_{idx}.md', io.BytesIO(content.strip().encode('utf-8')), 'text/markdown'))
+            print(f"\nProcessing row {idx}:")
+            print("Content being sent to API:")
+            print("-" * 50)
+            print(content)
+            print("-" * 50)
+            
+            # Extract metadata using dynamic extraction
+            try:
+                extracted_data = extract_structured_content(
+                    content,
+                    document_type="spreadsheet_entry"
+                )
+                
+                # Add detailed logging of the extraction results
+                print(f"\n=== Extraction Results for Row {idx} ===")
+                print(f"Content: {content.strip()}")
+                print("Extracted Metadata:")
+                print(json.dumps(extracted_data, ensure_ascii=False, indent=2))
+                print("=" * 50)
+                
+                # Create metadata structure
+                metadata = {
+                    'extracted_at': datetime.now(timezone.utc).isoformat(),
+                    'status': 'processed',
+                    'context_type': 'google-spreadsheet',
+                    'context_name': context_name,
+                    'source_content': content,
+                    'document_type': 'spreadsheet_entry',
+                    'extracted_data': extracted_data
+                }
+                
+                # Save metadata file
+                filename = f'{context_name}_{idx}.md'
+                metadata_path = Path(f'specs/{bot_id}/extraction/metadata') / f"{filename}.metadata.json"
```

Comment:
Generally I think this is not a great design - there's no need to store these json files in the filesystem. It's better to append the metadata as a fourth element in the `file_streams` list, instead of implicitly assuming the existence of a `.metadata.json` file somewhere.

Reply:
ok, changed,